roachpb/data.go

```diff
@@ -546,6 +546,9 @@ func (t *Transaction) Update(o *Transaction) {
 		*t = *proto.Clone(o).(*Transaction)
 		return
 	}
+	if len(t.Key) == 0 {
+		t.Key = o.Key
+	}
 	if o.Status != PENDING {
 		t.Status = o.Status
 	}
```

Comment:
could you re-enable the assertion in l. 575 below and remove the TODO? #2300 is done.